api/v1/awsloadbalancercontroller_types.go

```diff
@@ -64,10 +64,12 @@ type AWSLoadBalancerControllerSpec struct {
 	// will be propagated to the AWS resources. The controller owns all the tags of the managed AWS resources,
 	// unsolicited tags are removed. The controller doesn't watch for changes on AWS, so the removal of the unsolicited
 	// tags can only be triggered by an event coming from OpenShift. AWS supports a maximum of 50 tags per resource.
-	// AWSLoadBalancerController reserves 3 tags for its use, the rest is split between the tag annotation
-	// which can be set on the ingress and this field: 23 and 24, respectively. Each tag key must be unique.
+	// AWSLoadBalancerController reserves 3 tags for its use, OpenShift InfrastructureStatus reserves 25 tags,
+	// the rest is split between the tag annotation which can be set on the ingress and this field: 11 and 11,
+	// respectively. Each tag key must be unique. Tags from this field take precedence over those in InfrastructureStatus
+	// if a tag key conflicts.
 	//
-	// +kubebuilder:validation:MaxItems=24
+	// +kubebuilder:validation:MaxItems=11
```

Comment:
@deads2k: Can you please give us your opinion on this API change?

Context: `AWSLoadBalancerController` API provides this field to add additional resource tags to any ELB provisioned by the controller. Also, users can add resource tags for each individual ELB using dedicated annotations on a `Service` or an `Ingress`. With this PR we are adding the third party to the game - the platform tags. There is one constraint - AWS accepts only 50 tags on any resource.
 
I have two main concerns:
- The support of existing `AWSLoadBalancerController` resources which may have `>11` additional resource tags.
- The distribution of the 50 tags limit between the 3 parties. The current distribution is: `25` from `Infrastructure` CR, 11 from `AWSLoadBalancerController` CR and 11 from `Service` or `Ingress`.

Reply:
The schema cannot change this way and https://github.com/openshift/crd-schema-checker should be updated to prevent it.  Making it impossible it to set other values isn't a change we can make.  We have to live with 24 in the schema.

Reply:
Slack thread: https://redhat-internal.slack.com/archives/CE4L0F143/p1733165984170569.

Reply:
As per the discussion, API changes are reverted